msal/tests/Test.MSAL.NET.Unit/PublicClientApplicationTests.cs

```diff
@@ -1087,5 +1087,87 @@ public void GetUserTest()
             Assert.AreEqual(userToFind.Identifier, fetchedUser.Identifier);
             Assert.AreEqual(userToFind.Name, fetchedUser.Name);
         }
+
+        [TestMethod]
+        [Description("Test for AcquireToken with user canceling authentication")]
+        public async Task AcquireTokenWithAuthenticationCanceledTestAsync()
+        {
+            PublicClientApplication app = new PublicClientApplication(TestConstants.ClientId);
+
+            // Interactive call and user cancels authentication
+            MockWebUI ui = new MockWebUI()
+            {
+                MockResult = new AuthorizationResult(AuthorizationStatus.UserCancel,
+                    TestConstants.AuthorityHomeTenant + "?error=user_canceled")
+            };
+
+            //add mock response for tenant endpoint discovery
+            HttpMessageHandlerFactory.AddMockHandler(new MockHttpMessageHandler
+            {
+                Method = HttpMethod.Get,
+                ResponseMessage = MockHelpers.CreateOpenIdConfigurationResponse(TestConstants.AuthorityHomeTenant)
+            });
+
+            MsalMockHelpers.ConfigureMockWebUI(ui);
+
+            try
+            {
+                AuthenticationResult result = await app.AcquireTokenAsync(TestConstants.Scope).ConfigureAwait(false);
+            }
+            catch (MsalClientException exc)
+            {
+                Assert.IsNotNull(exc);
+                Assert.AreEqual("authentication_canceled", exc.ErrorCode);
```

Comment:
>ErrorCode [](start = 63, length = 9)

aside: imo the error code should be enum's - do you recall why it was agreed this should be a string?

Reply:
no...no idea why it's like this.